domains/auth/Auth.API/Tests/Controllers/TestTokenController.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using API.Configuration;
+using API.Models;
+using API.Services;
+using API.TokenStorage;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+using Xunit.Categories;
+
+namespace API.Controllers;
+
+[UnitTest]
+public class TestTokenController
+{
+    private readonly Mock<ICookies> cookiesMock = new();
+    private readonly Mock<IOptions<AuthOptions>> authOptionsMock = new();
+    private readonly Mock<ITokenStorage> tokenStorage = new();
+
+    private readonly TokenController tokenController;
+
+    public TestTokenController()
+    {
+        authOptionsMock.Setup(x => x.Value).Returns(new AuthOptions
+        {
+            CookieName = "Authorization",
+        });
+
+        tokenController = new TokenController(
+            authOptionsMock.Object,
+            tokenStorage.Object,
+            cookiesMock.Object
+        )
+        {
+            ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext()
+            }
+        };
+    }
+
+    [Fact]
+    public void ForwardAuth_OpaqueTokenAndInternalTokenExists_ReturnAuthorizationHeaderAndStatusCode200()
+    {
+        tokenStorage.Setup(x => x.GetInteralTokenByOpaqueToken(It.IsAny<string>())).Returns(new InternalToken
+        {
+            Actor = "Actor",
+            Subject = "Subject",
+            Scope = new List<string> { "Scope1", "Scope2" },
+            Issued = DateTime.UtcNow.AddHours(-1),
+            Expires = DateTime.UtcNow.AddHours(6)
+        });
+
+        var opaqueToken = "TestOpaqueToken";
+
+        tokenController.HttpContext.Request.Headers.Add("Cookie", $"Authorization={opaqueToken}");
+
+        var response = tokenController.ForwardAuth();
+
+        Assert.Equal(200, ((StatusCodeResult)response).StatusCode);
+    }
+
+    [Fact]
+    public void ForwardAuth_MissingOpaqueTokenInHeader_ReturnStatusCode401()
+    {
+        var response = tokenController.ForwardAuth();
+
+        Assert.Equal(401, ((StatusCodeResult)response).StatusCode);
+    }
+
+    [Fact]
+    public void ForwardAuth_OpaqueTokenExistsInHeaderNoInternalToken_ReturnStatusCode401()
+    {
+        var tokenStorage = new Mock<ITokenStorage>();
```

Comment:
Consider using:

// Arrange
// Act
// Assert

Reply:
I would actually prefer not to have comments in the code.

Reply:
Okay, that's a different discussion then. But I find that *some* way of separating these steps in a unit test helps the immediate overview and readability of it.

Reply:
We havent used it in this project at all, so i would prefer not to add it for now.

Reply:
If you look at the Emissions and Measurements API's, you will find them on most tests. I do think we should align our way of structuring tests.